src/document/code.rs

```diff
@@ -196,7 +197,7 @@ impl Line {
                             &self.indent,
                             comment_start,
                             if idx == 0 { &block_start } else { &block_next },
-                            path,
+                            path.display(),
```

Comment:
This does have a conceptual issue: While the result of `.display()` is safe to format and display to humans, it is not guaranteed to be lossless, i.e. it should be avoided if we want to parse this path back later.

There are various ways to go about this, but generally speaking, enforcing valid UTF-8 for paths that are "part of the project structure" seems a reasonable approach in this.